moped-editor/src/components/GridTable/FiltersChips.js

```diff
@@ -93,6 +93,23 @@ const FiltersChips = ({
     }
   };
 
+  /**
+   * Triggered by Filter Chip any/all click
+   * Gets previous IsOr state and sets the IsOr state and the url params state to be the opposite
+   */
+  const toggleIsOrOnClick = () => {
+    let newIsOrState = null;
```

Comment:
i'm trying to think through this because i'm concerned that assigning this variable in the `setIsOr` call could behave unpredictably since setting state is not synchronous.

would `const newIsOrState = !isOr;` work? I'm thinking that the two subsequent setter functions would be guaranteed to use the same value that way.

Reply:
thank you for thinking through this, I was initially going to have both setter functions just toggle the opposite of their previous states, but then i was concerned that they could get out of sync. 

I'll change it to the const !isOr, the state shouldn't set before that definition, so it should work?

Reply:
hm does this mean 
```
    const newIsOrState = !isOr;
    setIsOr(newIsOrState);
```
without referencing the prevState?

Reply:
yes, after looking at [the docs](https://react.dev/reference/react/Component#setstate), i don't think that `prevState` is needed since state is only being updated once. i'm also starting to wonder if just using `!isOr` is the way to go since its value won't update until the next render.